src/lib/linked-list.js

```diff
@@ -0,0 +1,132 @@
+'use strict';
+
+const Node = require('./node');
+
+module.exports = class LinkedList {
+    constructor() {
+        this.length = 0;
+        this.head = null;
+        this.tail = null;
+    }
+
+    push(value) { //Big O(n)
+        const node = new Node(value);
+        console.log(node);
+
+        if(!this.head) {
+            this.head = node;
+            this.tail = node;
+            this.length++;
+            return this;
+        }
+
+        let currentNode = this.head;
+        while(currentNode.next) {
+            currentNode = currentNode.next;
+        }
+        currentNode.next = node;
+        this.tail = currentNode.next;
+        this.length++;
+
+        return this;
+    }
+
+    pop() { //Big O(n)
+        if(this.length === 0) {
+            return undefined;
+        }
+
+        let currentNode = this.head;
+        if (!currentNode.next) {
+            this.head = null;
+            this.tail = null;
+            this.length--;
+            return currentNode;
+        }
+        while(currentNode.next) {
+            if (currentNode.next === this.tail) {
+                this.tail = currentNode;
+                currentNode.next = null;
+                this.length--;
+                return this;
+            }
+            currentNode = currentNode.next;
+        }
+
+        return this;
+    }
+
+    remove(value) { //Big O(n)
+        if (!this.head) {
+            return undefined;
+        }
+        let currentNode = this.head;
+        while(currentNode.next) {
+            if (currentNode.value === value) {
+                this.head = this.head.next;
+                this.length--;
+                return this;
+            }
+            if (currentNode.next.value === value) {
+                if (!currentNode.next.next) {
+                    this.tail = currentNode;
+                }
+                currentNode.next = currentNode.next.next;
+                this.length--;
+                return this;
+            }
+            currentNode = currentNode.next;
+        }
+
+        return this;
+    }
+
+};
+
+    Object.prototype.map = function(callback) { //Big O(n)
+    if (this === null) { //Linked List cannot be null
```

Comment:
Good checks for null and "not a function" checks, but best to return some kind of message so that it's clear to someone else executing code how this method works, i.e. throw an error that clearly tells a user "Linked List cannot be null" as opposed to just returning undefined.